modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/ComplimentaryDate.jsx

```diff
@@ -67,6 +79,125 @@ const ComplimentaryDate = ({
 		return dateLimitExceeded;
 	}, [startDate]);
 
+	const isComplementaryKeys = state.activationKeys?.map((item) => {
+		return item.complimentary;
+	});
+	const isComplementaryKey = [...new Set(isComplementaryKeys)].join(', ');
+
+	const hasDesiredEntry = state.activationKeys.some(
+		(item) =>
+			item.licenseEntryType === 'oem' ||
+			item.licenseEntryType === 'virtual-cluster' ||
+			item.licenseEntryType === 'enterprise'
+	);
+
+	const submitKey = useCallback(async () => {
+		const selectedFields = [
+			'active',
+			'description',
+			'hostName',
+			'ipAddresses',
+			'licenseEntryType',
+			'macAddresses',
+			'maxClusterNodes',
+			'name',
+			'productName',
+			'productVersion',
+			'sizing',
+		];
+
+		const saveSubscriptionKey = async (id) => {
+			return putSubscriptionInKey(provisioningServerAPI, id, sessionId);
+		};
+
+		const generateLicenseKey = async (item, isComplementary = false) => {
+			const licenseKey = {
+				accountKey,
+				complimentary: 'true',
+				expirationDate: endDate,
+				productKey: infoSelectedKey.selectedSubscription.productKey,
+				startDate,
+			};
+			selectedFields.forEach((field) => {
+				licenseKey[field] = item[field];
+			});
+			const response = await provisioningService.createNewGenerateKey(
+				accountKey,
+				licenseKey
+			);
+			if (checkedBoxSubscription && isComplementary) {
+				await saveSubscriptionKey(response?.items?.[0]?.id);
+			}
+		};
+
+		setIsLoadingGenerateKey(true);
+
+		try {
+			if (hasDesiredEntry) {
+				const createKeyPromises = state.activationKeys.map(
+					async (item) => {
+						await generateLicenseKey(item);
+					}
+				);
+
+				await Promise.all(createKeyPromises);
+				setIsLoadingGenerateKey(false);
+
+				return true;
+			} else {
+				const results = await Promise.all(
+					state.activationKeys.map(async (item) => {
+						await generateLicenseKey(item, isComplementaryKey);
+					})
+				);
+
+				await Promise.all(results);
+				setIsLoadingGenerateKey(false);
+				navigate(urlPreviousPage, {
```

Comment:
consistent spacing between `await Promise` and `setIsLoadingGenerateKey` and `navigate`